libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt

```diff
@@ -52,9 +53,22 @@ internal class TripSessionLocationEngine constructor(
         locationEngine.removeLocationUpdates(locationEngineCallback)
     }
 
+    private fun logIfLocationIsNotFreshEnough(location: Location) {
+        val currentTime = SystemClock.elapsedRealtimeNanos()
+        val locationAge = (currentTime - location.elapsedRealtimeNanos)
+        if (locationAge > DELAYED_LOCATION_WARNING_THRESHOLD_NS) {
+            logD(
+                TAG,
+                "Got obsolete location: current time = $currentTime, " +
+                    "elapsedRealtimeNanos = ${location.elapsedRealtimeNanos}"
+            )
+        }
+    }
```

Comment:
After some testing I realised that it's hard to read messages in logs. Here's my suggestion:
```suggestion
    private fun logIfLocationIsNotFreshEnough(location: Location) {
        val currentTime = SystemClock.elapsedRealtimeNanos()
        val locationAgeNanoSeconds = currentTime - location.elapsedRealtimeNanos
        val locationAgeMilliseconds = TimeUnit.MILLISECONDS.convert(
            locationAgeNanoSeconds,
            TimeUnit.NANOSECONDS
        )
        if (locationAgeMilliseconds > DELAYED_LOCATION_WARNING_THRESHOLD_MS) {
            logW(TAG, "Got an obsolete location: age = $locationAgeMilliseconds ms")
        }
    }
```

Reply:
Went ahead and rebase from `main` to fix the conflicts and applied this suggestion. BTW any particular reason to change from `logD` to `logW`?

Reply:
I've already applied it manually(left `logD` though).